fvm/environment/unsafe_random_generator_test.go

```diff
@@ -1,36 +1,64 @@
 package environment_test
 
 import (
+	"fmt"
+	"math"
+	mrand "math/rand"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"gonum.org/v1/gonum/stat"
 
 	"github.com/onflow/flow-go/fvm/environment"
 	"github.com/onflow/flow-go/fvm/tracing"
 	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
 )
 
 func TestUnsafeRandomGenerator(t *testing.T) {
-	t.Run("UnsafeRandom doesnt re-seed the random", func(t *testing.T) {
-		bh := &flow.Header{}
-
+	// basic randomness test to check outputs are "uniformly" spread over the
+	// output space
+	t.Run("randomness test", func(t *testing.T) {
+		bh := unittest.BlockHeaderFixtureOnChain(flow.Mainnet.Chain().ChainID())
 		urg := environment.NewUnsafeRandomGenerator(tracing.NewTracerSpan(), bh)
 
-		// 10 random numbers. extremely unlikely to get the same number all the time and just fail the test by chance
-		N := 10
-
-		numbers := make([]uint64, N)
+		sampleSize := 80000
+		tolerance := 0.05
+		n := 10 + mrand.Intn(100)
+		distribution := make([]float64, n)
 
-		for i := 0; i < N; i++ {
-			u, err := urg.UnsafeRandom()
+		// partition all outputs into `n` classes and compute the distribution
+		// over the partition. Each class is `classWidth`-big
+		classWidth := math.MaxUint64 / uint64(n)
+		// populate the distribution
+		for i := 0; i < sampleSize; i++ {
+			r, err := urg.UnsafeRandom()
 			require.NoError(t, err)
-			numbers[i] = u
+			distribution[r/classWidth] += 1.0
 		}
+		stdev := stat.StdDev(distribution, nil)
+		mean := stat.Mean(distribution, nil)
+		assert.Greater(t, tolerance*mean, stdev, fmt.Sprintf("basic randomness test failed. stdev %v, mean %v", stdev, mean))
+	})
 
-		allEqual := true
-		for i := 1; i < N; i++ {
-			allEqual = allEqual && numbers[i] == numbers[0]
+	// tests that unsafeRandom is PRG based and hence has deterministic outputs.
+	t.Run("PRG-based UnsafeRandom", func(t *testing.T) {
```

Comment:
can you add a test to verify each transaction with the block has a deterministic / unique seed?